src/pages/UserPage/UserPage.jsx

```diff
@@ -1,96 +1,204 @@
+      return {
+        id_address: address.id_address,
+        address_name: address.address_name,
+        address: formData.address[index].address,
+        city: formData.address[index].city,
+      };
+    });
+  
+    // creates new user obj with the information from the form
     const updatedUser = {
+      user_id: user.user_id,
       name,
+      last_name,
       email,
       password,
-      phoneNumber,
-      address,
-      city
-    }
+      phone_numbers: updatedPhoneNumbers,
+      addresses: updatedAddresses,
+      city,
+    };
+    
     const emailRegEx = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/;
     const passwordRegEx = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/g;
-    
-    if (name.length <= 1) {
-      validationErrors.name = 'name must be at least characters long';
+  
+    // validates the fields
+    if (!name.length) {
+      validationErrors.name = languageSelector(language, 'signInFirstNameError');
     }
-    if (emailRegEx.test(email)) {
-      validationErrors.email = 'Please enter a valid email';
+    if (!emailRegEx.test(email)) {
+      validationErrors.email = languageSelector(language, 'signInEmailError');
     }
-    if (passwordRegEx.test(password)) {
-      validationErrors.password = 'Please enter a valid password';
+    if (!passwordRegEx.test(password)) {
+      validationErrors.password = languageSelector(language, 'signInPasswordError');
     }
-    if (phoneNumber.length < 10) {
-      validationErrors.phoneNumber = 'Please enter a phone number that is at least 10 digits long';
+    if (phone_numbers.some((phoneNumber) => phoneNumber.phone_number.length < 10)) {
+      validationErrors.phoneNumber = languageSelector(language, 'signInPhoneError');
     }
-    if (address.length <= 2) {
-      validationErrors.address = 'Please enter a valid address';
+    if (address.some((addressItem) => !addressItem.address.trim())) {
+      validationErrors.address = languageSelector(language, 'userAddressError');
     }
-
     
-    if (Object.keys(validationErrors).length > 0) {
+    if (Object.keys(validationErrors).length) {
       setErrors(validationErrors);
       return;
     }
-
-    setUser(updatedUser);
-    setIsEditable(false);
-    setErrors({});
-  } 
+  
+    // sends the information to the back end and does a new petition
+    try {
+      await axios.put(`http://localhost:8080/api/users/${id}`, updatedUser);
+      dispatch(updateUserSuccess(updatedUser));
+      setIsEditable(false);
+      setErrors({});
+      const response = await axios.get(`http://localhost:8080/api/users/${id}`);
+      dispatch(fetchUserSuccess(response.data.data));
+      setFormData({
+        name: response.data.data.name,
+        last_name: response.data.data.last_name,
+        email: response.data.data.email,
+        password: response.data.data.password,
+        phone_numbers: response.data.data.phone_numbers,
+        address: response.data.data.addresses,
+        city: response.data.data.city
```

Comment:
as I mentioned before, destructure a better naming of the objects, for better understanding and maintenance.

Reply:
resolved, but changing address causes the app to crash